Unable to compile for ESP32-C3

### The problem

I'm trying to compile the base ESPHome image for the **ESP32-C3-DevKitM-1** but getting an error every time. I've tried different configurations but it results in a error every time. Also, I'm using the 'Manual Download' option to create the BIN file.

### Which version of ESPHome has the issue?

v2021.9.1

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2021.9.3

### What platform are you using?

ESP32

### Board

ESP32-C3-DevKitM-1

### Component causing the issue

firmware.bin

### Example YAML snippet

```yaml
esphome:
  name: esp32-c3
  platform: ESP32
  board: esp32-c3-devkitm-1
  platformio_options:
    platform: https://github.com/platformio/platform-espressif32#feature/arduino-idf-master
```


### Anything in the logs that might be useful for us?

```txt
Compiling /data/esp32-c3/.pioenvs/esp32-c3/lib049/Crypto/AES128.cpp.o
In file included from /root/.platformio/lib/Crypto/AES128.cpp:23:
/root/.platformio/lib/Crypto/AES.h:200:10: fatal error: hwcrypto/aes.h: No such file or directory
 #include "hwcrypto/aes.h"
          ^~~~~~~~~~~~~~~~
compilation terminated.
*** [/data/esp32-c3/.pioenvs/esp32-c3/lib049/Crypto/AES128.cpp.o] Error 1
========================= [FAILED] Took 586.29 seconds =========================
```


### Additional information

I've added the complete error log in a pastebin - https://pastebin.com/Fy5Ype71

--- a/esphome/components/airthings_wave_plus/sensor.py
+++ b/esphome/components/airthings_wave_plus/sensor.py
@@ -1,6 +1,7 @@
 import esphome.codegen as cg
 import esphome.config_validation as cv
 from esphome.components import sensor, ble_client
+from esphome.core import CORE
 
 from esphome.const import (
     DEVICE_CLASS_CARBON_DIOXIDE,
@@ -116,3 +117,6 @@ async def to_code(config):
     if CONF_TVOC in config:
         sens = await sensor.new_sensor(config[CONF_TVOC])
         cg.add(var.set_tvoc(sens))
+
+    if CORE.is_esp32:
+        cg.add_library("ESP32 BLE Arduino", None)

--- a/esphome/components/captive_portal/__init__.py
+++ b/esphome/components/captive_portal/__init__.py
@@ -3,7 +3,7 @@ import esphome.config_validation as cv
 from esphome.components import web_server_base
 from esphome.components.web_server_base import CONF_WEB_SERVER_BASE_ID
 from esphome.const import CONF_ID
-from esphome.core import coroutine_with_priority
+from esphome.core import coroutine_with_priority, CORE
 
 AUTO_LOAD = ["web_server_base"]
 DEPENDENCIES = ["wifi"]
@@ -32,3 +32,7 @@ async def to_code(config):
     var = cg.new_Pvariable(config[CONF_ID], paren)
     await cg.register_component(var, config)
     cg.add_define("USE_CAPTIVE_PORTAL")
+
+    if CORE.is_esp32:
+        cg.add_library("DNSServer", None)
+        cg.add_library("WiFi", None)

--- a/esphome/components/esp32/__init__.py
+++ b/esphome/components/esp32/__init__.py
@@ -276,6 +276,8 @@ async def to_code(config):
     cg.add_define("ESPHOME_BOARD", config[CONF_BOARD])
     cg.add_build_flag(f"-DUSE_ESP32_VARIANT_{config[CONF_VARIANT]}")
 
+    cg.add_platformio_option("lib_ldf_mode", "off")
+
     conf = config[CONF_FRAMEWORK]
     if conf[CONF_TYPE] == FRAMEWORK_ESP_IDF:
         cg.add_platformio_option(

--- a/esphome/components/ethernet/__init__.py
+++ b/esphome/components/ethernet/__init__.py
@@ -123,3 +123,6 @@ async def to_code(config):
         cg.add(var.set_manual_ip(manual_ip(config[CONF_MANUAL_IP])))
 
     cg.add_define("USE_ETHERNET")
+
+    if CORE.is_esp32:
+        cg.add_library("WiFi", None)

--- a/esphome/components/http_request/__init__.py
+++ b/esphome/components/http_request/__init__.py
@@ -92,6 +92,11 @@ async def to_code(config):
     cg.add(var.set_useragent(config[CONF_USERAGENT]))
     if CORE.is_esp8266 and not config[CONF_ESP8266_DISABLE_SSL_SUPPORT]:
         cg.add_define("USE_HTTP_REQUEST_ESP8266_HTTPS")
+
+    if CORE.is_esp32:
+        cg.add_library("WiFiClientSecure", None)
+        cg.add_library("HTTPClient", None)
+
     await cg.register_component(var, config)
 
 

--- a/esphome/components/mcp3008/__init__.py
+++ b/esphome/components/mcp3008/__init__.py
@@ -2,6 +2,7 @@ import esphome.codegen as cg
 import esphome.config_validation as cv
 from esphome.components import spi
 from esphome.const import CONF_ID
+from esphome.core import CORE
 
 DEPENDENCIES = ["spi"]
 AUTO_LOAD = ["sensor"]
@@ -23,3 +24,6 @@ async def to_code(config):
     var = cg.new_Pvariable(config[CONF_ID])
     await cg.register_component(var, config)
     await spi.register_spi_device(var, config)
+
+    if CORE.is_esp32:
+        cg.add_library("SPI", None)

--- a/esphome/components/mdns/__init__.py
+++ b/esphome/components/mdns/__init__.py
@@ -30,15 +30,16 @@ CONFIG_SCHEMA = cv.All(
 
 
 async def to_code(config):
-    if config[CONF_DISABLED]:
-        return
-
-    cg.add_define("USE_MDNS")
     if CORE.using_arduino:
         if CORE.is_esp32:
             cg.add_library("ESPmDNS", None)
         elif CORE.is_esp8266:
             cg.add_library("ESP8266mDNS", None)
+
+    if config[CONF_DISABLED]:
+        return
+
+    cg.add_define("USE_MDNS")
 
     var = cg.new_Pvariable(config[CONF_ID])
     await cg.register_component(var, config)

--- a/esphome/components/ota/__init__.py
+++ b/esphome/components/ota/__init__.py
@@ -101,6 +101,7 @@ async def to_code(config):
     if CORE.is_esp8266:
         cg.add_library("Update", None)
     elif CORE.is_esp32 and CORE.using_arduino:
+        cg.add_library("Update", None)
         cg.add_library("Hash", None)
 
     use_state_callback = False

--- a/esphome/components/spi/__init__.py
+++ b/esphome/components/spi/__init__.py
@@ -10,7 +10,7 @@ from esphome.const import (
     CONF_SPI_ID,
     CONF_CS_PIN,
 )
-from esphome.core import coroutine_with_priority
+from esphome.core import coroutine_with_priority, CORE
 
 CODEOWNERS = ["@esphome/core"]
 spi_ns = cg.esphome_ns.namespace("spi")
@@ -45,6 +45,9 @@ async def to_code(config):
     if CONF_MOSI_PIN in config:
         mosi = await cg.gpio_pin_expression(config[CONF_MOSI_PIN])
         cg.add(var.set_mosi(mosi))
+
+    if CORE.is_esp32:
+        cg.add_library("SPI", None)
 
 
 def spi_device_schema(cs_pin_required=True):

--- a/esphome/components/web_server_base/__init__.py
+++ b/esphome/components/web_server_base/__init__.py
@@ -24,6 +24,8 @@ async def to_code(config):
     await cg.register_component(var, config)
 
     if CORE.is_esp32:
+        cg.add_library("WiFi", None)
         cg.add_library("FS", None)
+        cg.add_library("Update", None)
     # https://github.com/esphome/ESPAsyncWebServer/blob/master/library.json
     cg.add_library("esphome/ESPAsyncWebServer-esphome", "1.3.0")

